contrib/signet/getcoins.py

```diff
@@ -5,18 +5,32 @@
 
 import argparse
 import subprocess
-import requests
 import sys
+import requests
+
+DEFAULT_GLOBAL_FAUCET = 'https://signetfaucet.com/claim'
+GLOBAL_FIRST_BLOCK_HASH = '00000086d6b2636cb2a392d45edc4ec544a10024d30141c9adf4bfd9de533b53'
 
 parser = argparse.ArgumentParser(description='Script to get coins from a faucet.', epilog='You may need to start with double-dash (--) when providing bitcoin-cli arguments.')
 parser.add_argument('-c', '--cmd', dest='cmd', default='bitcoin-cli', help='bitcoin-cli command to use')
-parser.add_argument('-f', '--faucet', dest='faucet', default='https://signetfaucet.com/claim', help='URL of the faucet')
+parser.add_argument('-f', '--faucet', dest='faucet', default=DEFAULT_GLOBAL_FAUCET, help='URL of the faucet')
 parser.add_argument('-a', '--addr', dest='addr', default='', help='Bitcoin address to which the faucet should send')
 parser.add_argument('-p', '--password', dest='password', default='', help='Faucet password, if any')
 parser.add_argument('bitcoin_cli_args', nargs='*', help='Arguments to pass on to bitcoin-cli (default: -signet)')
 
 args = parser.parse_args()
 
+if args.faucet.lower() == DEFAULT_GLOBAL_FAUCET:
+    # Get the hash of the block at height 1 of the currently active signet chain
+    try:
+        curr_signet_hash = subprocess.check_output([args.cmd] + args.bitcoin_cli_args + ['getblockhash', '1']).strip().decode()
+    except FileNotFoundError:
+        print('The binary', args.cmd, 'could not be found.')
+        exit()
+    if curr_signet_hash != GLOBAL_FIRST_BLOCK_HASH:
+        print('The global faucet cannot be used with a custom Signet network. Please use the global signet or setup your custom faucet to use this functionality.\n')
+        exit()
+
 if args.addr == '':
     if args.bitcoin_cli_args == []:
         args.bitcoin_cli_args = ['-signet']
```

Comment:
During the `getblockhash 1` RPC call `-signet` isn't set in `args.bitcoin_cli_args` yet. `bitcoin-cli` tries to reach the mainnet RPC port 8332. 

You need to set the `-signet` argument before you do the first RPC call. Probably just move the following up:
``` python
 if args.bitcoin_cli_args == []:
        args.bitcoin_cli_args = ['-signet']
```
        
(I guess this works fine if one sets `signet=1` in the `bitcoin.conf` file, but that can't be assumed to be the case)

Reply:
Absolutely! I guess I missed this edge case, thanks for mentioning! I've included this in the first commit and pushed the changes :)